plugins/techdocs-backend/src/service/router.ts

```diff
@@ -137,63 +151,60 @@ export async function createRouter({
           'https://backstage.io/docs/features/techdocs/architecture',
       );
     }
-
     // techdocs-backend will only try to build documentation for an entity if techdocs.builder is set to 'local'
-    // If set to 'external', it will only try to fetch and assume that an external process (e.g. CI/CD pipeline
-    // of the repository) is responsible for building and publishing documentation to the storage provider.
-    if (
-      config.getString('techdocs.builder') === 'local' &&
-      entity.metadata.uid &&
-      shouldCheckForUpdate(entity.metadata.uid)
-    ) {
-      const docsBuilder = new DocsBuilder({
-        preparers,
-        generators,
-        publisher,
-        dockerClient,
-        logger,
-        entity,
-      });
-      let foundDocs = false;
-      switch (publisherType) {
-        case 'local':
-        case 'awsS3':
-        case 'azureBlobStorage':
-        case 'openStackSwift':
-        case 'googleGcs':
-          // This block should be valid for all storage implementations. So no need to duplicate in future,
-          // add the publisher type in the list here.
-          await docsBuilder.build();
-          // With a maximum of ~5 seconds wait, check if the files got published and if docs will be fetched
-          // on the user's page. If not, respond with a message asking them to check back later.
-          // The delay here is to make sure GCS/AWS/etc. registers newly uploaded files which is usually <1 second
-          for (let attempt = 0; attempt < 5; attempt++) {
-            if (await publisher.hasDocsBeenGenerated(entity)) {
-              foundDocs = true;
-              break;
-            }
-            await new Promise(r => setTimeout(r, 1000));
-          }
-          if (!foundDocs) {
-            logger.error(
-              'Published files are taking longer to show up in storage. Something went wrong.',
-            );
-            res.status(408).json({
-              error:
-                'Sorry! It took too long for the generated docs to show up in storage. Check back later.',
-            });
-            return;
+    // If set to 'external', it will ssume that an external process (e.g. CI/CD pipeline
```

Comment:
```suggestion
    // If set to 'external', it will assume that an external process (e.g. CI/CD pipeline
```

Reply:
Currently in master the existing `navigate() `call on line 168 causes the doc to update without reloading page.
If I replace it with `history.pushState()` the url in address bar updates but the doc does not update.
It seems `pushState()` updates the URL but the `const { '*': path } = useParams();` doesn't pick up the change if done this way.
Yes, oddly enough this seems to contradict the fix in #4041.

It looks like the `if (parsedURl.hash)` part of the if condition isn't being matched so the existing pushState call isn't being used?